node/resolvers/checkout/index.ts

```diff
@@ -31,6 +35,20 @@ const shouldUpdateMarketingData = (orderFormMarketingTags, segmentData: SegmentD
     || utmiCampaign !== utmi_campaign)
 }
 
+const syncOrderFormAndSessionAddress = async (orderFormAddress, orderFormId, sessionAddress, ctx) => {
+  const {dataSources: {session, checkout}} = ctx
+
+  if (!orderFormAddress && sessionAddress) {
+    return checkout.updateOrderFormShipping(orderFormId, { clearAddressIfPostalCodeNotFound: false, selectedAddresses: [sessionAddress] })
```

Comment:
Why return here and await over there?

Reply:
Return early here with the new order form, with the new address.

its useful here:
```
const newOrderForm = await syncOrderFormAndSessionAddress(...)
return newOrderForm || orderForm
```

If it doest need to update the orderForm, `newOrderForm` will be null and we can just return the original `orderForm` we got before.

Do you think the await on line 47 is useless?